### PR TITLE
Faculty view

### LINKED ISSUE #82: Faculty View
### Get
**Data(/):**  
_About himself_
- id
- department
- user
  - id
  - name
  - email
  - contact 

**Table(/approvals):**
_About all the approvals he is related to (overview data only)_
- id
- status
- remarks
- salary
   - id
   - Salary Period
   - total CTC
   - job
      - id
      - role
      - company
         - id
         - name
      - season
         - id
         - year
         - type  
        
**Enums(/enums)**
- FacultyApprovalStatusEnum

### Update : Patch
**Change the status of a faculty Approval**
- id
- status

**Update Details About Self**
- id
- user
   - name
   - email
   - contact

### PR DESCRIPTION
**Pull request by AryanGKulkarni**
Closes #82
Closes #95

GET
- [x] Data about himself
- [x] Approvals
- [x] Enums

PATCH  
- [x] Change status of Faculty Approval
- [x] Update Details About Self

### CODE REVIEW

File: src/faculty-view/dto/get.dto.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+class SalaryDto {
+  @NestedUUID({})
+  id: string;
+
+  @NestedNumber({})
+  totalCTC: number;
+
+  @NestedObject({ type: JobDto })
+  job: JobDto;
```
Comment (cse210001015) [inline src/faculty-view/dto/get.dto.ts:70]:
```suggestion
  @NestedUUID({})
  id: string;

  @NestedNumber({})
  totalCTC: number;
  
  @NestedNumber({})
  totalCTC: number;
  
  @NestedString({ optional: true })
  salaryPeriod?: string;

  @NestedObject({ type: JobDto })
  job: JobDto;
```

File: src/faculty-view/dto/patch.dto.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { NestedEmail, NestedEnum, NestedObject, NestedString, NestedUUID } from "src/decorators/dto";
+import { FacultyApprovalStatusEnum } from "src/enums";
+
+export class UpdateFacultyApprovalStatusDto {
+  @NestedUUID({})
+  id: string;
+
+  @NestedEnum(FacultyApprovalStatusEnum, { optional: false })
+  status?: FacultyApprovalStatusEnum;
+}
```
Comment (cse210001015) [inline src/faculty-view/dto/patch.dto.ts:10]:
```suggestion
  @NestedEnum(FacultyApprovalStatusEnum, {})
  status: FacultyApprovalStatusEnum;
}
```

File: src/faculty-view/dto/patch.dto.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { NestedEmail, NestedEnum, NestedObject, NestedString, NestedUUID } from "src/decorators/dto";
+import { FacultyApprovalStatusEnum } from "src/enums";
+
+export class UpdateFacultyApprovalStatusDto {
+  @NestedUUID({})
+  id: string;
+
+  @NestedEnum(FacultyApprovalStatusEnum, { optional: false })
+  status?: FacultyApprovalStatusEnum;
```
Comment (cse210001015) [inline src/faculty-view/dto/patch.dto.ts:9]:
```suggestion
  @NestedEnum(FacultyApprovalStatusEnum, {})
  status: FacultyApprovalStatusEnum;
```

File: src/faculty-view/dto/query.dto.ts
Comment (cse210001015) [inline src/faculty-view/dto/query.dto.ts:1]: Isme approvals ke liye query honi chahiye not for faculty.

File: src/faculty-view/faculty-view.controller.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  @Get("enums")
+  async getEnums() {
+    const ans = await this.facultyViewService.getEnums();
+
+    return ans;
+  }
+
+  @Patch("approval-status")
+  async updateApprovalStatus(@Body() approval: UpdateFacultyApprovalStatusDto) {
+    const result = await this.facultyViewService.updateApprovalStatus(approval);
```
Comment (web-team-iiti) [inline src/faculty-view/faculty-view.controller.ts:48]:
```suggestion
    return await this.facultyViewService.updateApprovalStatus(approval);
```

File: src/faculty-view/faculty-view.controller.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  @ApiResponse({ type: FacultyApprovalRequestsDto, isArray: true })
+  async getFacultyApprovals(@Query("q") where: FacultyApprovalsQueryDto, @User() user: IUser) {
+    const ans = await this.facultyViewService.getApprovals({ facultyId: user.facultyId });
+
+    return pipeTransformArray(ans, FacultyApprovalRequestsDto);
+  }
+
+  @Get("enums")
+  async getEnums() {
+    const ans = await this.facultyViewService.getEnums();
```
Comment (web-team-iiti) [inline src/faculty-view/faculty-view.controller.ts:41]:
```suggestion
    return await this.facultyViewService.getEnums();
```

File: src/faculty-view/dto/get.dto.ts
Comment (web-team-iiti) [inline src/faculty-view/dto/get.dto.ts:1]: we can add API property here. though not necessary right now. but if we can

File: src/faculty-view/faculty-view.controller.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  async updateApprovalStatus(@Body() approval: UpdateFacultyApprovalStatusDto) {
+    const result = await this.facultyViewService.updateApprovalStatus(approval);
+
+    return result;
+  }
+
+  @Patch("faculty")
+  @UseInterceptors(TransactionInterceptor)
+  async updateFaculty(@Body() faculty: UpdateFacultyDto, @TransactionParam() t: Transaction) {
+    const result = await this.facultyViewService.updateFaculty(faculty, t);
```
Comment (mittal-ishaan) [inline src/faculty-view/faculty-view.controller.ts:56]:
```suggestion
    return await this.facultyViewService.updateFaculty(faculty, t);
```

**Review by gjain-7 — APPROVED** [review]

**Comment by gjain-7** [comment]
Remove Get enums faculty controller

**Pull request merged**